packages/devextreme/testing/testcafe/helpers/testAccessibility.ts

```diff
@@ -0,0 +1,94 @@
+import createWidget, { WidgetName } from './createWidget';
+import { a11yCheck, A11yCheckOptions, ElementContext } from './accessibilityUtils';
+import { isMaterialBased } from './themeUtils';
+
+export interface Options {
+  [key: string]: any[];
+}
+
+export interface Configuration {
+  component: WidgetName;
+  options?: Options;
+  a11yCheckConfig?: A11yCheckOptions;
+  selector?: ElementContext;
+  created?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+  after?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+}
+
+const defaultSelector = '#container';
+const defaultOptions = {};
+const defaultCreated = async () => {};
+const defaultAfter = async () => {};
+const defaultA11yCheckConfig = isMaterialBased() ? {
+  runOnly: 'color-contrast',
+} : {};
+
+const generateConfigurations = (
+  options: Options,
+  index = 0,
+  prevConfigurations: Options[] = [],
+) => {
+  const keys = Object.keys(options);
+
+  if (index >= keys.length) {
```

Comment:
We have `index === keys.length` as a base for recursion, `index` cannot be greater

```suggestion
  if (index === keys.length) {
```

Reply:
Fixed